### PR TITLE
rename object store to kv store

### PR DESCRIPTION
**Pull request by dennismartensson**
we are renaming the object store to kv store.

### CODE REVIEW

**Comment by fgsch** [comment]
Should we deprecate it for a version or two as to not break existing code?

File: _examples/objectstore/main.go
Comment (dgryski) [inline _examples/objectstore/main.go:1]: This enclosing directory should be renamed.

File: _examples/objectstore/main.go [hunk 1]
```diff
@@ -8,12 +8,12 @@ import (
 	"strings"
 
 	"github.com/fastly/compute-sdk-go/fsthttp"
-	"github.com/fastly/compute-sdk-go/objectstore"
+	kvstore "github.com/fastly/compute-sdk-go/kvstore"
```
Comment (dgryski) [inline _examples/objectstore/main.go:11]: Don't need a named import here -- package name and import name match.

**Comment by dennismartensson** [comment]
We have agreed that sdk changes for this name change can be a breaking version update.

**Review by dgryski — CHANGES_REQUESTED** [review]
Please a forwarding stub package with a deprecation notice like I did for the `edgedict` rename.  Writing `semgrep` rules for the transition is optional.

**Comment by dgryski** [comment]
Do we need to release his SDK right away for this change?

**Comment by dennismartensson** [comment]
let me give it a go and see what I end up with 😄

**Review by dgryski — APPROVED** [review]
LGTM.

File: kvstore/kvstore.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+	b, err := io.ReadAll(e)
+	if err != nil {
+		return ""
+	}
+
+	e.validString = true
+
+	return string(b)
+
+}
```
Comment (peterbourgon) [inline kvstore/kvstore.go:34]:
```suggestion
	e.s = string(b)
	e.validString = true
	return e.s
}
```

File: kvstore/kvstore.go [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+)
+
+var ErrKeyNotFound = errors.New("kvstore: key not found")
+
+type Entry struct {
+	io.Reader
+
+	validString bool
+	s           string
+}
```
Comment (peterbourgon) [inline kvstore/kvstore.go:17]: Embedding an io.Reader in this way allows callers to treat Entry as an io.Reader directly, which doesn't seem to be correct, as methods like String consume from that Reader and mutate internal state in a way that callers aren't able to do.

This should probably be something like

```suggestion
type Entry struct {
	src  io.Reader
	read bool
	str  string
}
```

Comment (joeshaw) [reply kvstore/kvstore.go:17]: Opened a follow-up issue on this. https://github.com/fastly/compute-sdk-go/issues/40

**Pull request merged**